contrib/libzip/premake5.lua

```diff
@@ -16,6 +16,8 @@ project "zip-lib"
 
 	filter "system:windows"
 		defines { "_WINDOWS" }
+	filter { "system:windows", "action:gmake**" }
```

Comment:
Yeah, I don't like this strategy... I feel like mingw/cygwin might be systems, but I'm not entirely sure :/

Reply:
> I feel like mingw/cygwin might be systems

I disagree, is there anything stopping you from using either of these to target Android or Linux? I'd be more inclined to say that these should fall under `toolset`, or a subtype of `toolset`/`action`.

Reply:
Maybe... the distinction between mingw/cygwin and 'windows' is that they have a different runtime, different ABI, different libs. They are *definitely* a different platform than 'windows' (assuming an MSVC windows world).

I have often wanted to generate a makefile that uses MSVC because I want to share the same build system across platforms. I don't usually do it, but I feel like it should be possible to produce a makefile that supports the normal 'windows' ecosystem, in addition to mingw/cygwin.

Reply:
Yeah nar. It's definitely not a toolset thinking on it. Mingw/cygwin are undeniably different platforms... or 'systems'?
Building mingw from windows is no different than building android from windows. The entire pipeline is incompatible with 'windows' by premakes conventional assumptions.

Reply:
I really don't know enough about either of them to push for anything in particular, they just don't seem like `system`s to me.